plugins/candela/plugin_tests/candelaSpec.js

```diff
@@ -104,19 +104,21 @@ describe('Test the candela UI.', function () {
             expect($('#g-user-action-menu.open').length).toBe(0);
             $('.g-user-text>a:first').click();
         });
-        girderTest.waitForLoad();
+        waitsFor(function () {
+            return $('#g-user-action-menu.open').length === 1;
+        }, 'menu to open');
 
         runs(function () {
-            expect($('#g-user-action-menu.open').length).toBe(1);
             $('a.g-my-folders').click();
         });
-        girderTest.waitForLoad();
+        waitsFor(function () {
+            return $('g-folder-list-link').length > 0;
+        }, 'user folder list to load');
```

Comment:
Is this new `waitsFor` sufficient to fix the race condition, or are these other changes necessary?

Reply:
I'm not sure what is causing the failure exactly, as I only have logs to go by. I'm just trying to make things more robust in general.